raster-foundry/raster-foundry
# Add frontend support for datasource licenses

## Overview

This PR adds frontend support for adding licenses to datasources. 

~**Merge this PR *AFTER* Doobie transition and licenses API are both done.** <- to avoid adding a new migration for this feature flag while a current migration for Doobie is not merged yet.~

**Updates on Mar. 28th:**
* Hook up frontend and backend 🎉 
* Remove WIP tag
* **Note on testing**: make sure to test on your **own** datasource

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

![licenses](https://user-images.githubusercontent.com/16109558/36568725-0c9eb766-17f9-11e8-98b1-79d0848b74c4.gif)


### Notes

* Hold off merging this PR until the API is ready.
* The `datasourceLicense.service.js` only tries to mock the api call and will be updated after finishing the API. 
* Two `*.json` files will be deleted before merging


## Testing Instructions

* Go to datasource page
* Test the datasource license selection feature and see if licenses are logged correctly in the console.

Closes #2956
